src/probe/http_probe.rs

```diff
@@ -26,59 +31,83 @@ lazy_static! {
 }
 
 pub async fn call_endpoint(
-    http_method: &String,
+    http_method: &str,
     url: &String,
     input_parameters: &Option<ProbeInputParameters>,
+    sensitive: bool,
 ) -> Result<EndpointResult, Box<dyn std::error::Error + Send>> {
     let timestamp_start = Utc::now();
-    let (otel_headers, trace_id) = get_otel_headers();
+    let (otel_headers, cx, span_id, trace_id) =
+        get_otel_headers(format!("{} {}", http_method, url));
 
     let request = build_request(http_method, url, input_parameters, otel_headers)?;
+    let request_timeout = Duration::from_secs(input_parameters.as_ref().map_or(
```

Comment:
I think we can do

```
    let request_timeout = Duration::from_secs(
        input_parameters
            .as_ref()
            .and_then(|params| params.timeout_seconds)
            .unwrap_or(DEFAULT_REQUEST_TIMEOUT_SECS)
    );
```

Reply:
ooh, I love that! Was looking for something like `.and_then` but didn't find it at the time!

Reply:
Applied this in [ef89b57](https://github.com/prodzilla/prodzilla/pull/16/commits/ef89b57a23f9c0924ff2e642fb44d3059f287a9f)